test/e2e/framework.go

```diff
@@ -696,8 +703,76 @@ func (data *TestData) deployAntreaFlowExporter(ipfixCollector string) error {
 	return data.mutateAntreaConfigMap(nil, ac, false, true)
 }
 
+// deployFlowVisibilityClickHouse deploys ClickHouse operator and DB.
+func (data *TestData) deployFlowVisibilityClickHouse() (*PodIPs, error) {
+	err := data.createNamespace(flowVisibilityNamespace, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := wait.Poll(2*time.Second, 10*time.Second, func() (bool, error) {
+		rc, stdout, stderr, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s", flowVisibilityYML))
+		if err != nil || rc != 0 {
+			// ClickHouseInstallation CRD applied soon before actual CR, and
```

Comment:
Trying to better understand the corner case, could you expand a bit more on what the exact error is you've encountered before?

Reply:
It only applies to e2e version of flow-visibility manifest (generated with `hack/generate-manifest-flow-visibility.sh --e2e`), where the yaml file contains clickhouse operator stuff including related CRDs directly from their repo, together with our `ClickHouseInstallation` CR. 

The API server seemingly takes time to register `ClickHouseInstallation` definition, and sometimes when resource of kind `ClickHouseInstallation` is applied, even at the end of the deployment yaml, api server still errors out claiming the CR is unknown. 

In the deployment steps doc we're asking user to apply ch operator yaml in a standalone step. The wait in between typically should be enough, so there shouldn't be errors when the flow-visibility yaml is applied.

Reply:
Got it, thanks. Can we rephrase the comment by a little bit? e.g. "ClickHouseInstallation CRD applied too soon before ClickHouse Operator installation completed." etc. 

How often does it occur? If it happens a lot, will re-apply the YAML manifest by a couple of times be sure to give a successful run within the 10s timeout? Would it be better to separate the operator installation from other resources?

Reply:
Have done a couple sets of Kind e2e and Jenkins for now and with the timeout even see the same error again. The retry explicitly checks for `ClickHouseInstallation` wording in the error message (stdout or stderr depends on testing provider somehow), and won't wait time on other errors.

separating operator yaml complicates stuff a lot. single yaml file have the following benefits especially for jenkins jobs:
- a single place to keep track of and change image pull strategy from Always to IfNotPresent, and repo from docker hub to distribution harbor
- for test_vmc script, automated image repo and tag extraction directly from the deployment file, so that when the tags are updated for operator and clickhouse server, no extra e2e script editing is needed. The pre-run pulling and loading is necessary given the relatively large size of both operator container and actual server.
- to better align with the hack/generate workflow, where only one yaml file is spit out to stdout.

Moreover, I'm not seeing benefit of separating the operator yaml in e2e. there won't be errors for the operator part, but when we're applying the ClickHouseInstallation CR, there's still a chance of hitting the exact same error, and the retry is still needed. applying the same yaml multiple times here won't create multiple instances of resources, so I'm not seeing a drawback in doing so.

Reply:
Thanks. I would also assume that operator's READY status would be an adequate sign of k8s api server going to accept the CH installation CR, but it's also possible that they chose to report ready even without api server side to fully take in the CR definition.  Let's keep the workflow as-is for now, and go back to it in the future if needed.